### ISSUE TITLE
LibreFox has moved

### ISSUE THREAD

**Issue by BeatLink** [intika/Librefox#125]
Due to the halting of development on this repository, we have created a new, community run successor to LibreFox, called LibreWolf. Check out the new organization here! https://gitlab.com/librewolf-community

**Comment by SuperRobinHood** [comment]
LibreVixen (female fox). Or if Mozilla's still bothered by that, then maybe LibreHawk. Name's the least important though - should revive the project first.

**Comment by petKitsune** [comment]
Vixen sounds nice
Icevixen?

don't know about LibreHawk
something more like a rodent
the power not to be seen

'should revive the project first'
yup

**Comment by petKitsune** [comment]
Vixen sounds nice
Icevixen?

don't know about LibreHawk
something more like a chipmunk
the power not to be seen
Tamias  'Tamias sibiricus'

'should revive the project first'
yup

**Comment by Serkan-devel** [comment]
Keep in mind that there are Firefox forks, like Waterfox or Palemoon for a long time already

**Comment by shreyasminocha** [comment]
Which is why we need @intika to update us on the status of things. We could come up with alternate names, but we *need* to know what the issues are. In my opinion, an issue with naming is unlikely because as @Serkan-devel said, there are other similarly named forks. Unfortunately, we aren't hearing from @intika, *possibly* because they have been gagged by Mozilla. If only we could get an update from @intika…

Does anyone apart from @intika have push access to this repo? I know that @intika entrusted some of us (myself included) with access to the @Librefox organization, but the main repo remains in their name (I think @intika eventually revoked our push rights from some of the other repos on the org). I'm not sure how this might turn out, but here's an idea: **what if we create and maintain a fork of the main repo in the organisation**? The option to create a repo in the org's name seems to be working for me (although I'm not completely sure). I would love if there were some way of verifying whether @intika will be coming back to this project, but in spite of our attempts, we haven't heard from them. It's been 4 months since their last commit or visible interaction with this repository. They seem to be active on GitHub otherwise. Odds are they have been gagged by Mozilla, but then again, we are in the dark as to what copyright issues there might possibly be. If we *do* choose to take this path, we should probably do our research and make sure we don't fringe on Mozilla's copyright(s). If @intika does eventually, return to this repo (I hope they do), they could merge this repo with the fork or vice-versa.

Edit: @Serkan-Devel seems to have suggested [something similar in the past](https://github.com/intika/Librefox/issues/119#issuecomment-480585450).

**Comment by SuperRobinHood** [comment]
I wrote Intika an e-mail a long time ago asking about this, and he didn't reply. He's probably not coming back then. I mean couldn't he at least reply privately?

**Comment by shreyasminocha** [comment]
@SuperRobinHood Nothing's private on the internet, of course, but I get your point. I once thought of writing to them too, but didn't. 

> He's probably not coming back then.

*Sigh*

Let me create a fork.

**Comment by shreyasminocha** [comment]
I've created [a fork on the org](https://github.com/Librefox/Librefox). I've given @intika admin access, in case they return. I don't know the in's and out's of the project, so I'd like to give some other people write access so they can review PRs as well. How do I go about this?

**Comment by Serkan-devel** [comment]
One problem with directly forking a project is that Github won't show these forks in search results.
And search through files on that repo won't be possible either
![image](https://user-images.githubusercontent.com/16866642/57191607-cfaadd80-6f27-11e9-99e5-8fefb353e7c9.png)

**Comment by shreyasminocha** [comment]
@Serkan-devel Yes, and we can't add a link to the main repo either since no-one but @intika has push access to it.

**Comment by Serkan-devel** [comment]
Maybe one could clone the repo locally and push the fork as it's own repo, then migrate all the open issues from this repo to there

**Comment by shreyasminocha** [comment]
Let me [contact support and ask them to switch Librefox/Librefox to 'normal mode'](https://stackoverflow.com/a/16052845), disassociating the repo from the fork. Can't delete the repo and redo since "Organization members can't delete repositories".

**Comment by BoostRacerKing** [comment]
Why can not you not completely rewrite the program so make your own then you did not need @intika anymore he will not come back anymore. But I think it's a bit brazen of him that he can not give at least one answer to us where the problem is

The name LibreVixen does not sound bad

**Comment by gerroon** [comment]
If you can devevelop this, just mirror it and start with a brand new repo and code base. There is no problem with that because the original devs clearly not in a position to communicate.

**Comment by ghost** [comment]
I don't understand how DCMA or this gag order works. I mean, aren't they supposed to let those accused know what the infringing part is and tell them what to do with it... use as is, attribute, don't modify, stop completely or whatever!~ 
This happened to the Gadgetbridge repo: https://github.com/Freeyourgadget/Gadgetbridge ... it completely went offline because a notice was filed and github blocked the whole thing, issues section as well (someone else's program's screenshots were used to discuss potential UI/UX solutions). Took a while, but things were cleared up, screenshots removed and the repo was restored completely.

**Comment by BeatLink** [comment]
If we do create a completely new organization, could we do it on gitlab? I do like the sound of LibreVixen as well by the way.

**Comment by Serkan-devel** [comment]
Gitlab.com is still poprietary but that might be a possibility

**Comment by Serkan-devel** [comment]
@shreyasminocha issues are currently disabled on your fork

**Comment by shreyasminocha** [comment]
@Serkan-devel Fixed.

**Comment by BeatLink** [comment]
Gitlab.com is proprietary? It runs on gitlab CE as far as i know

**Comment by Serkan-devel** [comment]
I doubt it ever ran CE though
![image](https://user-images.githubusercontent.com/16866642/57303197-1aba2180-70dd-11e9-9f6d-4080427c733b.png)

**Comment by i2mo** [comment]
> LibreVixen (female fox).

Just want to bring up that "Vixen" (or Wixen, Wicksen) means "to wank" in german ;-)

**Comment by BoostRacerKing** [comment]
> > LibreVixen (female fox).
> 
> Just want to bring up that "Vixen" (or Wixen, Wicksen) means "to wank" in german ;-)

Ja aber schreibt man nicht Wichsen so ?

**Comment by Serkan-devel** [comment]
Ja, ebend

**Comment by ghost** [comment]
Vixen and Fork(en), LOL : )

Forking this project is fine, but, let's say things go smoothly for a few days/weeks... what is to say that Mozilla won't slap the DCMA/gag order again. So, that road will lead to the same place. My suggestion is to brainstorm a list of things that could have caused Mozilla to react in such a manner! Trying to identify the issue is very important... this is where everyone, our collective brains+research can help. Further, looking at other forks of Firefox and how those projects handled/avoided potential issues from our list could help! At least this way, we are creating a roadmap and systematically stepping through... because the road has mines (LOL) and we don't where they are!

**Comment by elypter** [comment]
I would try to avoid everything that could cause a tradmark issue but if mozilla sees "Librefox" as a threat then i assume they will try to go after anything they can. Mozilla doesnt want to protect their Brand but they want to kill projects that do not behave as they please. For that reason i want to suggest the option to move primary development over to a tor hidden service if that is feasable. not as a means to do something illegal but as an additional line of defense. a website and download on the normal web can be hosted by an independend party. this way the developers are safe and if the website is taken down it can easily be rehosted and mirrored somewhere else much like the Torproject itself.

**Comment by ghost** [comment]
#Trademark -- okay, but how? There are other forks and have been going strong for a while. 
#"... kill projects that do not behave ..." - Okay, and how are we not behaving? This is where we need to come up with more ideas on where, what and how Mozilla sees this project as a threat. For example: 

**Stronger security / privacy: but they are already cherry picking features from the TOR browser
**Integrating addons into FF itself, thus, making some addons redundant? 

... What else could be pissing off Mozilla?

What you say about going stealth... giving the finger to Mozilla (LOL) and following TOR project's workflow (develop and release) sounds okay, but, not if you don't have to.

**Comment by BeatLink** [comment]
Well, we could start by forking to Gitlab, and if anything, we can move to a self hosted instance once we have the resources. I've taken the liberty of going ahead and creating the repository here: https://gitlab.com/librevixen . Just message me with your usernames so I can add you.

As for the reasons firefox is angry, I'm not too sure. I believe firefox has a compile time option that should remove all trademarks and logos and so on. We could use that.

**Comment by gerroon** [comment]
See this fork of Iceweasel for Win, might offer some clue maybe?
https://github.com/muslayev/iceweasel-win64

**Comment by brainscar** [comment]
@BeatLink please add me @brainscar

**Comment by brainscar** [comment]
@theel0ja
@shreyasminocha
@nhynes
@yoasif
@bogachenko
@szepeviktor

Please give gitlab username to @BeatLink

**Comment by brainscar** [comment]
Thank you @intika for all your hard work.

**Comment by szepeviktor** [comment]
mine is the same as here https://gitlab.com/szepeviktor

**Comment by shreyasminocha** [comment]
@BeatLink Mine's `shreyasminocha`. Thanks.

**Comment by Serkan-devel** [comment]
What about the forked librefox repo though?
And why on Gitlab?

**Comment by ghost** [comment]
I suspected that Mozilla is trying to becomes a very privacy conscious browser... with the **TOR Uplift project**... especially when there is money (Research Grants) involved... but this may not be it, even though it looks like their goals might overlap with this project.  

https://www.ghacks.net/2019/05/09/firefox-might-get-a-super-private-browsing-mode-in-the-future/

Trademarks and logos issue is an easy fix. Mozilla's TOR Uplift project is a good project to watch.

**Comment by Serkan-devel** [comment]
Could development still continue [here](https://github.com/Librefox/Librefox)?

**Comment by ghost** [comment]
It is not a bad idea to have a GitLab mirror? (copy), just in case Github decides (is forced or asked) to block this project completely.

**Comment by BeatLink** [comment]
So guys, do you want to develop on Gitlab or here https://github.com/Librefox/Librefox? Thumbs up this message for Gitlab, down for here. In any case, whichever decision we choose, we should probably rename to Librevixen ASAP, and remove the "fox" from the name, as I believe that's what causing the issue. Also, I think the issue is also due to the fact that in the the readme, we portray ourselves as a modification of of firefox, and not an independent fork.

**Comment by Serkan-devel** [comment]
> we should probably rename to Librevixen ASAP

I'd switch to anything that doesn't want to place misspelled "freefap" as an icon on my desktop.
And even though I have a Gitlab account and can develop there, gitlab.com is just proprietary + freebait, hosted on Microsoft Azure (or they switched to google cloud already)

> just in case Github decides (is forced or asked) to block this project completely.

Gitlab.com blocked some blocklist-repo because the people on that list reported it, so it's not bulletproof either

**Comment by ghost** [comment]
@BeatLink Excellent ideas on what could be causing the issue and how to fix them! I too remember the project feeling like "a modification of firefox"... when I first found out about it. 

@Serkan-devel Yes, that is why some one needs to keep a copy offline as well.

**Comment by Serkan-devel** [comment]
I guess one can just git clone it.

**Comment by BeatLink** [comment]
So it's decided then? We'll continue development here (at least for now), on the fork that @shreyasminocha made. And if the worst comes to the worst, we can fall back on the Gitlab repository. So I'm going to go ahead and close this issue, so we can continue the discussion here: https://github.com/Librefox/Librefox/issues/1

**Comment by BeatLink** [comment]
Sorry to reopen the issue once again, and sorry for the confusion, Because the fork is still on the organization with the name "Librefox", we had to create a completely new organization called librevixen. It can be found here https://github.com/librevixen. Just message me here or https://github.com/librevixen/LibreVixen/issues/1 so we can add you to the organization. I'll also leave this issue open for future visitors.

**Comment by elypter** [comment]
i am also not happy about a name which means to wank in german

**Comment by BeatLink** [comment]
> i am also not happy about a name which means to wank in german

Name's been changed to LibreWolf!

**Comment by dimqua** [comment]
@BeatLink Why not move to GitLab or Gitea/Gogs instances like notabug.org?

**Comment by BeatLink** [comment]
> @BeatLink Why not move to GitLab or Gitea/Gogs instances like notabug.org?

We were thinking of doing that in the future. Our core issue right now, is finding an online CI/CD or build system. Both travis and gitlab CI/CD have timeouts that expire before the build can be completed. Without an online build service, everyone would have to compile LibreWolf from scratch.

**Comment by dimqua** [comment]
@BeatLink does building it from the source code have any benefits? Isn't it possible to just use [unbranded builds](https://wiki.mozilla.org/Add-ons/Extension_Signing#Unbranded_Builds) as a base?

**Comment by BeatLink** [comment]
> @BeatLink does building it from the source code have any benefits? Isn't it possible to just use [unbranded builds](https://wiki.mozilla.org/Add-ons/Extension_Signing#Unbranded_Builds) as a base?

Those unbranded builds don't use our own naming scheme. The executable's name is still firefox and the profile folder it uses is still .mozilla, which might cause conflicts with an existing installation

**Comment by dimqua** [comment]
You could rename the executable as you wish.
> the profile folder it uses is still .mozilla

Yes, but it uses its own profile folder: ![](https://user-images.githubusercontent.com/540898/59167622-c8847a00-8b3a-11e9-9b7a-3d5baca599e3.png)

**Comment by BeatLink** [comment]
Okay so I just downloaded it and tested it out. 

Attempting to change the executable name results in:
Exec failed with error: No such file or directory

And while a separate profile is created, it is still in the ~/.mozilla/firefox folder. which is problematic because we want to get as far away from Mozilla's branding as possible and to not have it affect an existing install of firefox in anyway. 

So i still think we need to do our own compilation. Plus it will help with other privacy improvements such as compiling without telemetry and pocket and the like

**Comment by dimqua** [comment]
Do I understand correctly, that is (mostly) because of a possibility of potential legal threats?

**Comment by BeatLink** [comment]
Yes. While I do not know the full details, i understand that the reason LibreFox originally went into a hiatus was some issues related to copyright. As a result, while we continue this fork in intika's absence, we will aim as much as possible to distance ourselves from Firefox's branding and existing installations, to minimize any possible complications. Incidentally, this is a community run continuation. Should intika contact us to rejoin the project, we are happy to have them return (I'm learning this as I go, if I'm being honest with you)

**Comment by BeatLink** [comment]
We have moved to gitlab here, https://gitlab.com/librewolf-community, due to several benefits, but mainly that we have found a way to use their CI/CD system with runners on our personal machines to compile librewolf. See you over there!

**Comment by dimqua** [comment]
@BeatLink are you able to build multiple versions for different platforms ATM?

**Comment by BeatLink** [comment]
> @BeatLink are you able to build multiple versions for different platforms ATM?

Just linux at the current moment

**Comment by r3pek** [comment]
@BeatLink 
might be too late 'cause it looks like you're all settle there on gitlab, but i don't mind hosting it on my private gitea instance that has drone ci in it.

**Comment by BeatLink** [comment]
Thank you for the offer! But yeah, you're right, we've just got settled on gitlab :).

If you're willing and able to, you can help us out by firing up a Gitlab runner for CI/CD! I could really use the help with compiling.

**Comment by filips123** [comment]
Why move to GitLab? This will create just more confusion and differences. You can also use some other CI/CD systems with GitHub.

**Comment by BeatLink** [comment]
We moved to Gitlab mainly because it has an integrated CI/CD system that allowed us to build and test LibreWolf. This CI/CD system also allows us to setup CI environments on our own personal devices, which allowed us to solve an issue Github couldn't: Almost all the CI/CD systems on Github has job timeouts that would be too short for the compile times of librewolf. By using CI/CD setups on our own systems, integrated into Gitlab's infrastructure, we could seamlessly build, test and deploy librewolf without this worry of timeouts.

Furthermore, Gitlab is open core and self hostable, should we require to do so and have the available infrastructure, unlike Github, who is not only proprietary but also owned by Microsoft. Gitlab also has an active community who believes in some of our ethos. 

I understand your misgivings about creating more confusion. Rest assured that Gitlab will be the final place of LibreWolf from now on.

**Comment by filips123** [comment]
> Furthermore, Gitlab is open core and self hostable, should we require to do so and have the available infrastructure, unlike Github, who is not only proprietary but also owned by Microsoft.

But gitlab.com (which you currently use) is hosted on Google Cloud. This is same bad as Microsoft (or even worse). Could you run self-hosted instance but keep CI/CD system?

**Comment by BeatLink** [comment]
We do not have the infrastructure to perform CI/CD builds much less run our own Gitlab instance. I do not see how an open source software running on a proprietary infrastructure is worse than a completely proprietary service owned and operated by microsoft.

**Comment by filips123** [comment]
> I do not see how an open source software running on a proprietary infrastructure is worse than a completely proprietary service owned and operated by microsoft.

It's not. It's still good if you can't run your own instance. I wanted to say that I would not trust Google anything more than Microsoft. Both of them are big USA corporations and both of them track users.

**Comment by r3pek** [comment]
@BeatLink do you hang out on IRC or XMPP where we can actually chat about providing you a CI environment? ;)

**Comment by BeatLink** [comment]
I do get on IRC sometimes. Where should we meet?

**Comment by r3pek** [comment]
@BeatLink I'm on Freenode most of the time. Just drop me a query ;)

**Comment by theel0ja** [comment]
> But gitlab.com (which you currently use) is hosted on Google Cloud. This is same bad as Microsoft (or even worse). Could you run self-hosted instance but keep CI/CD system?

You can use Gitlab CI on your own instance, but you need to provide the nodes yourself.

**Comment by XeonG** [comment]
Any active download releases for LibreWolf, seems the Gitlab pages are even worse to navigate and find anything than GitHub

**Comment by shreyasminocha** [comment]
@XeonG This is slightly late, but check the [Releases](https://gitlab.com/librewolf-community/browser/linux/-/releases) tab on the GitLab repo. We now have some preliminary releases thanks to [@/ohfp](https://gitlab.com/ohfp). The download button on https://librewolf-community.gitlab.io now works too.

[v73.0.1-1](https://gitlab.com/librewolf-community/browser/linux/-/releases/v73.0.1-1) is the latest working version.

**Comment by subnut** [comment]
@shreyasminocha How about keeping a mirror on codeberg.org? 

Since we are currently unable to host git ourselves, I think we should host the project on a community-owned website (e.g. codeberg, notabug, framagit, etc.) instead of GitLab (due to the concerns other users have mentioned above)

**Comment by adrelanos** [comment]
From https://github.com/intika/Librefox/commit/45a4d3ce647b8c896e81ce3e5ac5ddc95ff045e0 in 2019:

@shreyasminocha:
> What's the status of your conversation with Mozilla? How are we in violation of their terms (if we are)?

@intika:
> i met with @AnXh3L0 FOSDEM event was pretty cool in short we need to wait an official answer from mozilla so we need to wait a little bit... i will post tomorrow details about that

That tomorrow has never come.

@AnXh3L0 could you shed any light on this please?

Do you have any idea why @intika disappeared?

**Comment by adrelanos** [comment]
Was actually raised and replied here:
https://github.com/intika/Librefox/issues/130#issuecomment-581026665

**Comment by ioc2e3** [comment]
>  , seems the Gitlab pages are even worse to navigate and find anything than GitHub

Agree ...

It’s harder to navigate and find content on the page than github.com ...

.
